certbot/certbot/plugins/enhancements.py

```diff
@@ -130,7 +146,7 @@ def update_autohsts(self, lineage, *args, **kwargs):
         """
 
     @abc.abstractmethod
-    def deploy_autohsts(self, lineage, *args, **kwargs):
+    def deploy_autohsts(self, lineage: interfaces.RenewableCert, *args: Any, **kwargs: Any) -> None:
```

Comment:
Based on how `enhancements.enable` is typed, it seems likely that `lineage` here would end up being `Optional`.

Reply:
`deploy_autohsts` is not called by `enhancements.enable`. From the code it seems that `interfaces.RenewableCert` is the type for the `domain` passed to the function.